### ISSUE TITLE
Regression in v5.0.0-beta.9 or v5.0.0-beta.10 (SyntaxError: Invalid or unexpected token)

### ISSUE THREAD

**Issue by mcongy** [webpack/enhanced-resolve#242]
After upgrading webpack@5 from beta22, I'm getting following error:

```
node_modules\enhanced-resolve\lib\SyncAsyncFileSystemDecorator.js:1
SyntaxError: Invalid or unexpected token
    at wrapSafe (internal/modules/cjs/loader.js:1117:16)
    at Module._compile (internal/modules/cjs/loader.js:1165:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
    at Module.load (internal/modules/cjs/loader.js:1050:32)
    at Function.Module._load (internal/modules/cjs/loader.js:938:14)
    at Module.require (internal/modules/cjs/loader.js:1090:19)
    at require (internal/modules/cjs/helpers.js:75:18)
    at Object.<anonymous> (C:\Projects\core\node_modules\enhanced-resolve\lib\ResolverFactory.js:11:38)
    at Module._compile (internal/modules/cjs/loader.js:1201:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1221:10)
```

Webpack config:
```
const config = {
    node: false,
    target: 'node',
    mode,
    devtool: 'source-map',
    stats: 'errors-only',
    performance: { hints: false },
    entry: join(process.cwd(), entryFilePath),
    externals: [nodeExternals({})],
    optimization: {
      minimize: false,
      noEmitOnErrors: true
    },
    module: {
      rules: [
        {
          test: /\.(js|ts)$/,
          exclude: /node_modules/,
          use: {
            loader: 'babel-loader',
            options: ...
          }
        },
        {
          test: /\.(graphql|gql|txt|sql)?$/,
          exclude: /node_modules/,
          use: 'raw-loader'
        },
        {
          test: /\.mjs$/,
          include: /node_modules/,
          type: 'javascript/auto'
        }
      ]
    },
    resolve: {
      extensions: ['.ts', '.js', '.graphql', '.json', '.gql'],
      alias: resolveTsconfigPathsToAlias()
    },
    output: ...
};
```
Babel plugins:
@babel/preset-typescript
babel-plugin-transform-typescript-metadata
@babel/preset-env (target node: 12.18.3)

Probably a regression in v5.0.0-beta.9 or v5.0.0-beta.10

I might post this issue to webpack repo, if it makes more sense.
Let me know if there's anything else you need to know to solve this.

Thank you for looking into it!

**Comment by evilebottnawi** [comment]
Yes, please create reproducible test repo

**Comment by sokra** [comment]
This is pretty weird. Sounds like the file is corrupted. Try to delete `node_modules` and reinstall. Maybe this fixes the problem.

**Comment by evilebottnawi** [comment]
@mcongy feel free to feedback if reinstall is not help, also please create reproducible test repo and I reopen the issue

**Comment by mcongy** [comment]
@evilebottnawi @mcongy 
I'm no longer having the problem using webpack@beta30 and enhanced-resolve@beta-12.

Anyway, I tried reinstalling node_modules multiple times, and tried it on different platforms (mac, linux, windows) and had the same issue.

Since it's already working, I won't create a reproducible test repo (I tried, but the application that was being compiled had 1000s of files and I had no clue where the problem could be).
